clang/test/3C/macro_itype.c

```diff
@@ -0,0 +1,30 @@
+// RUN: rm -rf %t*
+// RUN: 3c -base-dir=%S -alltypes -addcr %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: 3c -base-dir=%S -addcr %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
+// RUN: 3c -base-dir=%S -alltypes -addcr %s -- | %clang -c -fcheckedc-extension -x c -o /dev/null -
+// RUN: 3c -base-dir=%S -alltypes -output-dir=%t.checked %s --
+// RUN: 3c -base-dir=%t.checked -alltypes %t.checked/macro_itype.c -- | diff %t.checked/macro_itype.c -
+
+// Example encountered while converting libjpeg. This triggered an assertion
+// fail because the ItypeStr extracted from the source was empty.
+#define macro0 int *a : itype(_Ptr<int>);
+struct s { macro0 };
+//CHECK: struct s { macro0 };
+
+// Just removing the assertion that failed on the above example caused this to
+// rewrite incorectly. The ItypeStr would be left empty, so first paramter
+// would be rewritten to `int *b` even though the rewriter intened to give it
+// an itype. If the paramter was then passed a checked pointer, there would a
+// Checked C compiler error. Ideally, 3C wouldn't need to change the
+// declaration of `b` at all, but this is the usual limitation we have with
+// rewriting around macros.
+#define macro1 : itype(_Ptr<int>)
+void fn(int *b macro1, int *c) {
+//CHECK: void fn(int *b : itype(_Ptr<int>), _Ptr<int> c) {
```

Comment:
Can you please add an itype to `clang/test/3C/params_in_macro.c` too?  I was going to include an itype there originally, but I was blocked by the assertion that you are now removing in this PR.

Reply:
updated. Let me know if that's not what you were thinking